### ISSUE TITLE
Win .exe very big size

### ISSUE THREAD

**Issue by ruben-cruz** [electron-userland/electron-builder#7980]
* **Electron-Builder Version**: 22.2.0

* **Node Version**: 16.15.1

* **Electron Version**: 7.1.9
* **Electron Type (current, beta, nightly)**: current

* **Target**: win portable

I'm building a small windows portable app to include inside a CD/DVDs and the generated .exe file is about 45mb. I need to generate a smaller .exe to increase the app start performance inside the CD.

Does anyone know if it's possible to generate smaller .exe portable files? If so, how can we do that?

My package.json only has the following dependencies:
```
"devDependencies": {
    "asar": "^3.2.0",
    "electron": "^7.1.9",
    "electron-builder": "^22.2.0",
    "electron-packager": "^14.2.0",
    "javascript-obfuscator": "^4.1.0"
  },
  "dependencies": {
    "electron-pdf-window": "^1.0.12",
    "fs-extra": "^8.1.0",
    "systeminformation": "^4.19.1",
    "v8-compile-cache": "^2.1.0"
  },
```

**Comment by mmaietta** [comment]
I can't support earlier versions of electron-builder due to the way the versioning/release CI works on this monorepo. Can you first try the most recent 24.6.x version of electron builder first?

AFAIK, portable exe files need to have all package files included. The alternative is to use nsis-web target, but then you lose the portable nature of the application.

Best guess as a workaround is to have your large assets stored elsewhere and either copied from the disc or downloaded from a server on first app launch.

**Comment by ruben-cruz** [comment]
@mmaietta I've updated the electron-builder to v24.9.1, then generated a new portable, but the .exe size remains the same (~45mb).

I think I understood your suggestion, but the problem is that my app can be opened using a computer without network/internet connection. So, the assets should be stored inside the cd/dvd.

Do you know if it's possible to generate a portable .exe with some assets outside the app? In order to decrease the app .exe size...

**Comment by tendermario** [comment]
I don't think this question/issue is really the scope of `electron-builder`. You might get somewhere asking on a subreddit, search StackOverflow [(here's an example answer)](https://stackoverflow.com/a/66466353) or direct it towards Electron themselves, but I think Electron is not the most performant or smallest package, and that's accepted sacrifice for ease of creating a simple to setup desktop program.

45 MB is not large, CDs and DVDs can hold this size of a file easily.

**Comment by yuripourre** [comment]
> @mmaietta I've updated the electron-builder to v24.9.1, then generated a new portable, but the .exe size remains the same (~45mb).
> 
> I think I understood your suggestion, but the problem is that my app can be opened using a computer without network/internet connection. So, the assets should be stored inside the cd/dvd.
> 
> Do you know if it's possible to generate a portable .exe with some assets outside the app? In order to decrease the app .exe size...

Yes, it's possible! If you are only targeting windows, you can get the absolute path (where your application is running) then you need to expose it through the contextBridge so when you load the files you use the path as a reference.

If you target AppImage you need to use the path from `process.env.APP_IMAGE` otherwise it will point to the temporary folder.

I hope it helps!
